aws/resource_aws_storagegateway_gateway.go

```diff
@@ -457,6 +501,27 @@ func resourceAwsStorageGatewayGatewayUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if d.HasChanges("average_download_rate_limit_in_bits_per_sec",
+		"average_upload_rate_limit_in_bits_per_sec") {
+
+		avgDownRateLimit, avgDownRateLimitOk := d.GetOk("average_download_rate_limit_in_bits_per_sec")
+		avgUpRateLimit, avgUpRateLimitOk := d.GetOk("average_upload_rate_limit_in_bits_per_sec")
+
+		if avgDownRateLimitOk && avgUpRateLimitOk {
```

Comment:
It seems like it should be possible to only update one of the values. 😄  Maybe splitting the testing would confirm. Also curious if its possible to remove rate limits once they are set. Otherwise that limitation should be documented.

Reply:
will check